### ISSUE TITLE
Adyen 2.4.5 PaymentHandlerImpl NullPointerException

### ISSUE THREAD

**Issue by ovy9086** [Adyen/adyen-android#84]
Using :
```
implementation 'com.adyen.checkout:core:2.4.5'
implementation 'com.adyen.checkout:ui:2.4.5'
```

We get random crashes from LIVE with the following stacktrace :

```
java.lang.NullPointerException Attempt to write to field 'boolean com.adyen.checkout.core.internal.persistence.PaymentInitiationResponseEntity.handled' on a null object reference 
    PaymentHandlerImpl.java:144 com.adyen.checkout.core.internal.PaymentHandlerImpl$1.onHandled
    BaseManager.java:47 com.adyen.checkout.core.internal.BaseManager.checkDispatch
    BaseManager.java:37 com.adyen.checkout.core.internal.BaseManager.setData
    PaymentHandlerImpl.java:458 com.adyen.checkout.core.internal.PaymentHandlerImpl$8.run
    Handler.java:873 android.os.Handler.handleCallback
    Handler.java:99 android.os.Handler.dispatchMessage
    Looper.java:193 android.os.Looper.loop
    ActivityThread.java:6863 android.app.ActivityThread.main
    Method.java:-2 java.lang.reflect.Method.invoke
    RuntimeInit.java:537 com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run
    ZygoteInit.java:858 com.android.internal.os.ZygoteInit.main
```

**Comment by caiofaustino** [comment]
This is the same issue as 85 right?
https://github.com/Adyen/adyen-android/issues/85

**Comment by ovy9086** [comment]
I don't think so. They where diferent crash groups in our bug tracker, and the stacktraces do not indicate to me that they would be the same issue , as they differ 🤔

**Comment by ovy9086** [comment]
so, any updates on this ?

**Comment by caiofaustino** [comment]
Sorry we are not able to further investigate the issue without more information. Do you have a specific scenario where that happens? Any more information other than the stack trace?

**Comment by ovy9086** [comment]
no. I get this from the live app crashlogs in BugSnag.

**Comment by andreservidonigyg** [comment]
Hello everyone, how is the status of this crash? We get here too.

**Comment by caiofaustino** [comment]
Hey @andreservidonigyg, do you have more information on the issue? What scenario it happens? More logs?

**Comment by andreservidonigyg** [comment]
> Hey @andreservidonigyg, do you have more information on the issue? What scenario it happens? More logs?

Hello @caiofaustino i have the same logs that ovy sent. But from our reports we are getting this from Samsung devices. The crash occurs when we are requesting for adyen session from PaymentController.handlePaymentSessionResponse(...)

 Currently, our app in production is using the version 2.3.3, but we are working to upgrade it to 2.4.5.

`Fatal Exception: java.lang.NullPointerException: Attempt to write to field 'boolean com.adyen.checkout.core.internal.persistence.PaymentInitiationResponseEntity.handled' on a null object reference
       at com.adyen.checkout.core.internal.PaymentHandlerImpl$1.onHandled + 140(PaymentHandlerImpl.java:140)
       at com.adyen.checkout.core.internal.BaseManager.checkDispatch + 47(BaseManager.java:47)
       at com.adyen.checkout.core.internal.BaseManager.setData + 37(BaseManager.java:37)
       at com.adyen.checkout.core.internal.PaymentHandlerImpl$8.run + 419(PaymentHandlerImpl.java:419)
       at android.os.Handler.handleCallback + 873(Handler.java:873)
       at android.os.Handler.dispatchMessage + 99(Handler.java:99)
       at android.os.Looper.loop + 214(Looper.java:214)
       at android.app.ActivityThread.main + 7050(ActivityThread.java:7050)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run + 493(RuntimeInit.java:493)
       at com.android.internal.os.ZygoteInit.main + 965(ZygoteInit.java:965)`

**Comment by caiofaustino** [comment]
Thanks! That's some good information for me to start digging. I'll try to reproduce it.

**Comment by andreservidonigyg** [comment]
Hey @caiofaustino any update on the progress? Thanks!

**Comment by caiofaustino** [comment]
Yeah, I found a possible root cause. I made a fix that should be available in the next release. I'll notify here when we make the release.

**Comment by andreservidonigyg** [comment]
> Yeah, I found a possible root cause. I made a fix that should be available in the next release. I'll notify here when we make the release.

Cool! Thanks a lot!

**Comment by andreservidonigyg** [comment]
@caiofaustino the next release will be a 2.4.6 or will be include in 3.X.X+?

**Comment by caiofaustino** [comment]
It will be 2.4.9, current latest version of the SDK is 2.4.8. Versions 3.x.x we call Components or Drop-in, not SDK.

**Comment by caiofaustino** [comment]
We just released version `2.4.9` with a fix. Please update see if it's solved. Feel free to reopen the issued if it still persists.
